### PR TITLE
Add missing transforms, clear iobinding

### PR DESCRIPTION
**Pull request by ashbhandare**
This PR enables some missing graph fusion transforms, which helps reduce peak activation memory size, and also clears output iobinding to remove ORT's ref to the gradient buffer.

### CODE REVIEW

File: onnxruntime/core/optimizer/graph_transformer_utils.cc [hunk 1]
```diff
@@ -151,6 +153,8 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
       transformers.emplace_back(onnxruntime::make_unique<AttentionFusion>(cpu_cuda_execution_providers));
       transformers.emplace_back(onnxruntime::make_unique<EmbedLayerNormFusion>(cpu_cuda_execution_providers));
 
+      transformers.emplace_back(onnxruntime::make_unique<BiasDropoutFusion>(cpu_cuda_execution_providers));
+      transformers.emplace_back(onnxruntime::make_unique<MatmulTransposeFusion>(cpu_cuda_execution_providers));
```
Comment (SherlockNoMad) [inline onnxruntime/core/optimizer/graph_transformer_utils.cc:157]: MatmulTransposeFusion is a duplication of MatMulScaleFusion at line 160. 
MatmulTransposeFusion should be deprecated, and we should fix MatMulScaleFusion and use it.

File: orttraining/orttraining/python/training/ortmodule.py [hunk 2]
```diff
@@ -293,6 +293,7 @@ def backward(ctx, *grad_output):
                 results = [torch.tensor([1])] * len(self._onnx_graphs_info.user_input_names)
                 results += [_ort_output_to_torch_tensor(backward_output) \
                     for backward_output in backward_outputs[:len(self._onnx_graphs_info.initializer_grad_names_to_train)]]
+                self._gradient_io_binding.clear_binding_outputs()
```
Comment (SherlockNoMad) [inline orttraining/orttraining/python/training/ortmodule.py:296]: Add a comment explaining, the ORTValue is a shared pointer and this call only destroy the IOBinding instance of ORTValue, there is another instance managed by DLTensor.

File: onnxruntime/core/optimizer/graph_transformer_utils.cc (hunk 1 above)
Comment (SherlockNoMad) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc:157]: https://msdata.visualstudio.com/Vienna/_sprints/taskboard/ONNX%20Training/Vienna/Cobalt?workitem=1028960

Comment (ashbhandare) [reply onnxruntime/core/optimizer/graph_transformer_utils.cc:157]: adding a TODO for now to fix and combine the transforms

File: orttraining/orttraining/python/training/ortmodule.py (hunk 2 above)
Comment (ashbhandare) [reply orttraining/orttraining/python/training/ortmodule.py:301]: done

**Review by SherlockNoMad — APPROVED** [review]
